### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by andrewconner**

### CODE REVIEW

File: shoebox/app/com/keepit/model/Comment.scala [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  externalId: ExternalId[Comment] = ExternalId(),
+  normalizedURI: Id[NormalizedURI],
+  userId: Id[User],
+  text: String,
+  permissions: State[Comment.Permission] = Comment.Permissions.PUBLIC,
+  state: State[Comment] = Comment.States.ACTIVE
+) {
+  def withState(state: State[Comment]) = copy(state = state)
+  def withPermissions(permissions: State[Comment.Permission]) = copy(permissions = permissions)
+  def withExternalId(externalId: ExternalId[Comment]) = copy(externalId = externalId)
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:27]: why would you need that?

File: shoebox/app/com/keepit/model/Comment.scala [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  def withNormalizedURI(normalizedURI: Id[NormalizedURI]) = copy(normalizedURI = normalizedURI)
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:28]: and that?

File: shoebox/app/com/keepit/model/Comment.scala (hunk 1 above)
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:27]: test

File: shoebox/app/com/keepit/model/Comment.scala [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    val entity = CommentEntity(this.copy(updatedAt = currentDateTime))
+    assert(1 == entity.save())
+    entity.view
+  }
+
+}
+
+object Comment {
+
+  def apply(url: String, userId: Id[User], text: String, title: String, permissions: State[Comment.Permission])(implicit conn: Connection): Comment = {
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:40]: no implicit creation or query of other objects please

File: shoebox/app/com/keepit/model/Comment.scala [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    val nuri = NormalizedURI.getByNormalizedUrl(url) match {
+      case Some(nuri) =>
+        nuri
+      case None =>
+        NormalizedURI(title, url)
```
Comment (eishay) [inline shoebox/app/com/keepit/model/Comment.scala:45]: test!

**Comment by eishay** [comment]
magic hands!

File: shoebox/app/com/keepit/model/Comment.scala (hunk 2 above)
Comment (andrewconner) [inline shoebox/app/com/keepit/model/Comment.scala:28]: I need to get all comments for normalized URI, so this is valuable.

**Pull request merged**